wrappers/src/fdw/cognito_fdw/cognito_client/row.rs

```
@@ -0,0 +1,80 @@
+use supabase_wrappers::prelude::Cell;
+use supabase_wrappers::prelude::Column;
+use supabase_wrappers::prelude::Row;
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct ResultPayload {
+    pub(crate) users: Vec<CognitoUser>,
+    pub(crate) next_page_offset: Option<u64>,
+}
+
+#[derive(Debug)]
+pub struct CognitoFields(HashMap<String, Value>);
+
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CognitoUser {
+    pub created_at: String,
+    pub email: String,
+    pub email_verified: bool,
+    pub identities: Option<serde_json::Value>,
+    // Additional fields from UserType
+    pub username: String,
+    pub status: Option<String>,
+}
+
+pub trait IntoRow {
+    fn into_row(self, columns: &[Column]) -> Row;
+}
+
+impl IntoRow for UserType {
+    fn into_row(self, columns: &[Column]) -> Row {
+        let mut row = Row::new();
+
+        for column in columns {
+            match column.name.as_str() {
+                "username" => {
+                    if let Some(ref username) = self.username {
+                        row.push("username", Some(Cell::String(username.to_string())));
+                    }
+                }
+                "created_at" => {
+                    if let Some(created_at) = self.extract_attribute_value("created_at") {
+                        row.push("created_at", Some(Cell::String(created_at)));
+                    }
+                }
+                // TODO: update columns
+                "email" => {
+                    if let Some(email) = self.extract_attribute_value("email") {
+                        row.push("email", Some(Cell::String(email)));
+                    }
+                }
+                "status" => {
+                    if let Some(status) = self.extract_attribute_value("status") {
+                        row.push("status", Some(Cell::String(status)));
+                    }
+                }
+                _ => (), // Ignore unknown columns or handle them appropriately
```

we can return an error here to tell user this column isn't supported. For example, https://github.com/supabase/wrappers/blob/42f50957ed5411ec4fee1996509d11fa3d68f3f1/wrappers/src/fdw/mssql_fdw/mssql_fdw.rs#L69